compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -719,6 +719,18 @@ abstract class PrepJSInterop extends plugins.PluginComponent
                 "A Scala.js-defined JS trait cannot contain private members")
           }
         }
+
+        // @ScalaJSDefined overrides with different js names - issue #1983
+        sym.allOverriddenSymbols.headOption match {
```

Comment:
There's no reason to limit this test to non-native JS classes. I suggest to apply it to *all* JS types (native and Scala.js-defined).